src/headless/converse-chatboxes.js

```diff
@@ -1191,13 +1190,6 @@ converse.plugins.add('converse-chatboxes', {
                     );
                 }
 
-                const bare_forward = sizzle(`message > forwarded[xmlns="${Strophe.NS.FORWARD}"]`, stanza).length;
-                if (bare_forward) {
-                    return this.rejectMessage(
-                        stanza,
-                        'Forwarded messages not part of an encapsulating protocol are not supported'
-                    );
-                }
```

Comment:
This is dangerous, because if the `forward-message` plugin is not loaded, then removing this code causes a security vulnerability which allows malicious users to spoof messages as if they're coming from someone else.

I'm not sure currently what the right way is to go about adding this feature. One option is to make message forwarding a more "core" feature such as message corrections and spoilers, meaning that it doesn't go into its own plugin but instead goes into `converse-chatboxes` and `converse-chatview`. However I'm not yet convinced that this is the right way.

I've been thinking about using [JXT] from @legastero to convert incoming XML stanzas to JSON. I think if we did that, we could let the `forward-message` plugin register a JXT definition for message forwarding. That way, if the plugin is not loaded, forwarded messages are dropped because they can't be parsed and we avoid this problem. That's at least the theory.

Reply:
Hm. I've been thinking about this. I haven't tested this, but would it be possible to register our own strophe.js-handler (http://strophe.im/strophejs/doc/1.1.3/files/strophe-js.html#Strophe.Connection.addHandler) inside the plugin? That way the handler should not be registered if the plugin is missing and forwarded messages should be dropped again. When the plugin is active, the handler takes the stanza and calls the new functions. If it is not a forwarded message it will be dropped (I don't know if strophe.js allows to register a handler just for XEP-0297-stanzas). 
I can't test this right now, but I think this could be a solution without relying on JXT. What do you think?

Reply:
It doesn't look like you can register a handler only for forwarded messages, however you could register a more broader handler and then inside the handler check if the message is a forwarded one.